src/common/i2c.rs

```diff
@@ -0,0 +1,170 @@
+#![allow(dead_code)]
+
+use super::config::{AttributeError, Kind};
+use std::sync::{Arc, Mutex};
+
+// A trait representing blocking I2C communication for a board. TODO: replace with the
+// embedded_hal I2C trait when supporting boards beyond ESP32. AddressType is
+// either u8 (indicating support for 7-bit addresses) or u16 (for supporting 10-bit addresses)
+pub trait I2CHandle<AddressType> {
```

Comment:
Don't think we need to have the address type be a template parameter, it's going to be whatever is implementing the trait that is going to make the distinction between 10/7 at the config/build level